packages/netlify-cms-core/src/components/Editor/EditorToolbar.js

```diff
@@ -593,6 +599,14 @@ export class EditorToolbar extends React.Component {
       >
         {isPersisting ? t('editor.editorToolbar.saving') : t('editor.editorToolbar.save')}
       </SaveButton>,
+      currentStatus
+        ? [
+            this.renderWorkflowStatusControls(),
+            this.renderNewEntryWorkflowPublishControls({ canCreate, canPublish }),
+          ]
+        : !isNewEntry
+        ? this.renderExistingEntryWorkflowPublishControls({ canCreate, canPublish, canDelete })
```

Comment:
This is quite hard to follow.

Do you think we can replace the second ternary operator with `!isNewEntry && this.renderExistingEntryWorkflowPublishControls`?